### PR TITLE
SW-4813 Implemented hooks for planting site creation

### PR DESCRIPTION
**Pull request by tommylau523**

### CODE REVIEW

**Comment by tommylau523** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/terraware/terraware-web/2723?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#2723** <a href="https://app.graphite.dev/github/pr/terraware/terraware-web/2723?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#2720** <a href="https://app.graphite.dev/github/pr/terraware/terraware-web/2720?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @tommylau523 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: src/scenes/PlantingSitesRouter/hooks/usePlantingSiteCreate.ts [hunk 1]
```diff
@@ -18,25 +24,51 @@ export type Response = {
...
+      snackbar.toastError(strings.GENERIC_ERROR);
+    }
+  }, [createResult?.status, deleteDraftResult?.status, snackbar]);
+
+  useEffect(() => {
+    if (createResult?.status === 'success' && createRequest) {
+      const dispatched = dispatch(requestDeleteDraft(createRequest.id));
+      setDeleteDraftRequestId(dispatched.requestId);
+    }
+  }, [createRequest, createResult?.status, snackbar]);
```
Comment (nickgraz) [inline src/scenes/PlantingSitesRouter/hooks/usePlantingSiteCreate.ts:58]: I think `snackbar` can be removed here

File: src/scenes/PlantingSitesRouter/hooks/usePlantingSiteCreate.ts [hunk 2]
```diff
@@ -18,25 +24,51 @@ export type Response = {
...
+
+  useEffect(() => {
+    if (deleteDraftResult?.status === 'success' && createResult.data) {
+      goToPlantingSiteView(createResult.data);
+    }
+  }, [deleteDraftResult, createResult, goToPlantingSiteView, snackbar]);
```
Comment (nickgraz) [inline src/scenes/PlantingSitesRouter/hooks/usePlantingSiteCreate.ts:64]: Same here

**Review by nickgraz — APPROVED** [review]
Thanks for splitting up the functionality into hooks. One small nitpick about use effect dependencies but otherwise looks great.

**Comment by tommylau523** [comment]
### Merge activity

* **Jun 11, 8:56 PM EDT**: [Graphite](https://app.graphite.dev/github/pr/terraware/terraware-web/2723) rebased this pull request as part of a merge.

**Pull request merged**